scripts/globals/helm.lua

```diff
@@ -1496,6 +1496,34 @@ xi.helm.initZone = function(zone, helmType)
     end
 end
 
+xi.helm.result = function(player, helmType, broke, itemID)
+    local zoneId = player:getZoneID()
+
+    -- Egg-Hunt Extravaganza Event
+    if xi.events and xi.events.eggHunt then
+        xi.events.eggHunt.helmResult(player, itemID)
+    end
+
+    -- Quest: Vanishing Act
+    if
+        helmType == xi.helm.type.HARVESTING and
+        player:getQuestStatus(xi.quest.log_id.AHT_URHGAN, xi.quest.id.ahtUrhgan.VANISHING_ACT) == QUEST_ACCEPTED and
+        not player:hasKeyItem(xi.ki.RAINBOW_BERRY) and
+        broke ~= 1 and
+        zoneId == xi.zone.WAJAOM_WOODLANDS
+    then
+        npcUtil.giveKeyItem(player, xi.ki.RAINBOW_BERRY)
+    end
+
+    -- Missiom: AMK04
+    if xi.settings.main.ENABLE_AMK == 1 then
+        xi.amk.helpers.helmTrade(player, helmType, broke)
+    end
+
+    -- Records of Eminence
+    player:triggerRoeEvent(xi.roeTrigger.HELM_SUCCESS, { ['skillType'] = helmType })
```

Comment:
Wouldnt this change make it so the RoE is ALWAYS triggered no matter if you failed or not?

Reply:
Good catch, I've fixed.